### PR TITLE
Add Max Size Tag for Protobuf Fields

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds the max size tag for the protobuf fields and reformatted fields to `ssz-max` and `ssz-size`

### CODE REVIEW

File: proto/beacon/p2p/v1/types.proto [hunk 1]
```diff
@@ -13,54 +13,54 @@ message BeaconState {
 
   // History [2001-3000]
   BeaconBlockHeader latest_block_header = 2001;
-  repeated bytes block_roots = 2002 [(gogoproto.moretags) = "ssz:\"size=8192,32\""];
-  repeated bytes state_roots = 2003 [(gogoproto.moretags) = "ssz:\"size=8192,32\""];
-  repeated bytes historical_roots = 2004 [(gogoproto.moretags) = "ssz:\"size=?,32\""];
+  repeated bytes block_roots = 2002 [(gogoproto.moretags) = "ssz-size:\"8192, 32\""];
+  repeated bytes state_roots = 2003 [(gogoproto.moretags) = "ssz-size:\"8192, 32\""];
+  repeated bytes historical_roots = 2004 [(gogoproto.moretags) = "ssz:\"size=?,32, max=16777216\""];
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto]: What about this one?

File: proto/beacon/p2p/v1/types.pb.go [hunk 2]
```diff
@@ -10,6 +10,7 @@ import (
 
 	_ "github.com/gogo/protobuf/gogoproto"
 	proto "github.com/gogo/protobuf/proto"
+	github_com_prysmaticlabs_go_bitfield "github.com/prysmaticlabs/go-bitfield"
```
Comment (nisdas) [inline proto/beacon/p2p/v1/types.pb.go]: can we change the aliasing here ? its pretty ugly

Comment (prestonvanloon) [reply proto/beacon/p2p/v1/types.pb.go]: This is generated code. It's ok for it to be ugly

**Comment by prestonvanloon** [comment]
Some tests are failing due to ssz treehash not yet supporting bitlist/bitvector4

File: proto/beacon/p2p/v1/types.proto (hunk 1 above)
Comment (terencechain) [reply proto/beacon/p2p/v1/types.proto]: thanks for fixing it!

File: proto/beacon/p2p/v1/types.proto [hunk 3]
```diff
@@ -69,13 +69,13 @@ message PendingAttestation {
...
-  AttestationData data = 1;
   // Bitfield representation of validator indices that have voted exactly
   // the same vote and have been aggregated into this attestation.
   // Spec type: Bitlist[4096]
-  bytes aggregation_bits = 2 [(gogoproto.moretags) = "ssz-max:\"4096\"", (gogoproto.casttype) = "github.com/prysmaticlabs/go-bitfield.Bitlist"];
+  bytes aggregation_bits = 1 [(gogoproto.moretags) = "ssz-max:\"4096\"", (gogoproto.casttype) = "github.com/prysmaticlabs/go-bitfield.Bitlist"];
+  AttestationData data = 2;
   // Challengeable bit (SSZ-bool, 1 byte) for the custody of crosslink data. Not used in phase 0.
-  bytes custody_bits = 3 [(gogoproto.moretags) = "ssz-max:\"4096\""];
+  bytes custody_bits = 3 [(gogoproto.moretags) = "ssz-max:\"4096\"", (gogoproto.casttype) = "github.com/prysmaticlabs/go-bitfield.Bitlist"];
```
Comment (prestonvanloon) [inline proto/beacon/p2p/v1/types.proto]: Thanks! Missed this one.

**Comment by prestonvanloon** [comment]
@rauljordan can you comment on the status of go-ssz to support bitfields?

**Pull request merged**